### ISSUE TITLE
1.0.0.alpha1 upgrade post-mortem

### ISSUE THREAD

**Issue by mensfeld** [dry-rb/dry-validation#467]
Here are some problems I've encountered while trying to upgrade:

https://github.com/mensfeld/dry-problems

## Problem 1

While the i18n scope search works well when rule is defined like so: `rule(:kafka)` it does not work (or the scope is built in a way I don't understand) when nesting is provided:

```ruby
  rule(kafka: :ssl_ca_cert) do
    failure(:invalid)
  end
```

## Problem 2

As states in the example: when I define schema that is not met for nested structure, despite building rule with nesting, the rule is being evaluated even when schema is not met:

`bundle exec ruby problem2_execution_when_not_schema.rb`

## Problem 3

I18n scopes with rule name make custom rules potentially duplicate messages:

Assuming I have a method as linked in the problem3.rb file. When I create multiple rules with it and I decide to have the same error message, I will have to duplicate it between rules due to the way namespaces are being scoped.

## Problem 4

When there is no message in i18n the exception is not being raised with the path, making it hard to debug.
